core/src/test/scala/org/apache/spark/scheduler/BlacklistTrackerSuite.scala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import org.apache.spark.SparkFunSuite
+import org.scalatest.BeforeAndAfter
+import org.apache.spark.SparkConf
+import org.apache.spark.TaskEndReason
+import org.apache.spark.Success
+import org.apache.spark.ExceptionFailure
+import org.apache.spark.LocalSparkContext
+import org.apache.spark.SparkContext
+import org.mockito.Mockito.{when, spy}
+import org.apache.spark.util.ManualClock
+
+class BlacklistTrackerSuite extends SparkFunSuite with BeforeAndAfter with LocalSparkContext {
+
+  val FAILURE: TaskEndReason = new ExceptionFailure(
+      "Fake",
+      "fake failure",
+      Array.empty[StackTraceElement],
+      "fake stack trace",
+      None,
+      None)
+
+  val stage1 = 1
+  val stage2 = 2
+  // Variable name can indicate basic information of taskInfo
+  // The format is "taskInfo_executorId_taskIndex_hostName"
+  val taskInfo_1_1_hostA = new TaskInfo(1L, 1, 1, 0L, "1", "hostA", TaskLocality.ANY, false)
+  val taskInfo_1_2_hostA = new TaskInfo(2L, 2, 1, 0L, "1", "hostA", TaskLocality.ANY, false)
+  val taskInfo_2_1_hostA = new TaskInfo(3L, 1, 1, 0L, "2", "hostA", TaskLocality.ANY, false)
+  val taskInfo_2_2_hostA = new TaskInfo(4L, 2, 1, 0L, "2", "hostA", TaskLocality.ANY, false)
+  val taskInfo_3_3_hostB = new TaskInfo(5L, 3, 1, 0L, "3", "hostB", TaskLocality.ANY, false)
+
+  val clock = new ManualClock(0)
+
+  test ("expireExecutorsInBlacklist works") {
+    // expire time is set to 6s
+    val conf = new SparkConf().setAppName("test").setMaster("local")
+      .set("spark.ui.enabled", "false")
+      .set("spark.scheduler.executorTaskBlacklistTime", "6000")
+
+    sc = new SparkContext(conf)
+    val scheduler = new TaskSchedulerImpl(sc, 1)
```

Comment:
you can avoid creating a `SparkContext` in all of these tests with mockito.  add the `MockitoSugar` trait and then you can do:

```scala
val scheduler = mock[TaskSchedulerImpl]
when(scheduler.getExecutorsAliveOnHost(any[String])).thenReturn(Some(Set[String]()))
```
(or whatever else you you want that method to return).  Then you don't need to mixin `LocalSparkContext`.  it'll speed up the tests a bit